lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -227,6 +251,76 @@ public APIGatewayProxyResponseEvent handleRequest(
         }
     }
 
+    @Tracing
+    private void validateAndStoreHMRCInheritedIdentity(
+            String userId, JWTClaimsSet claimsSet, IpvSessionItem ipvSessionItem)
+            throws RecoverableJarValidationException, ParseException {
+        JarClaims claims;
+        try {
+            claims =
+                    OBJECT_MAPPER.convertValue(
+                            claimsSet.getJSONObjectClaim(CLAIMS_CLAIM), JarClaims.class);
+        } catch (IllegalArgumentException | ParseException e) {
+            throw buildRecoverableException(
+                    OAuth2Error.INVALID_REQUEST_OBJECT.setDescription("Unable to process claims"),
+                    claimsSet);
+        }
+        if (claims == null) {
+            return;
+        }
+
+        var userInfo = claims.userInfo();
+        if (userInfo == null) {
+            return;
+        }
+
+        var inheritedIdentityJwtClaim = userInfo.inheritedIdentityClaim();
+        if (inheritedIdentityJwtClaim == null) {
+            return;
+        }
+
+        SignedJWT signedInheritedIdentityJWT;
```

Comment:
Whenever I see this pattern it smells of needing to extract a method - if you have a method, you can return from within the block.

(In truth, I just hate any non-final variables :) )

Reply:
I've made it final by nesting a try/ catch, though could do this another way